src/core/thread/mlr_manager.hpp

```diff
@@ -113,8 +130,34 @@ class MlrManager : public InstanceLocator, public Notifier::Receiver
                                                      const Ip6::MessageInfo *aMessageInfo,
                                                      otError                 aResult);
 
-    uint16_t CountNetifMulticastAddressesToRegister(void) const;
-    void     SetNetifMulticastAddressMlrState(MlrState aFromState, MlrState aToState);
+#if OPENTHREAD_CONFIG_MLR_ENABLE
+    void UpdateLocalSubscriptions(void);
+    void SetNetifMulticastAddressMlrState(MlrState aFromState, MlrState aToState);
+    bool IsAddressMlrRegisteredByNetif(const Ip6::Address &aAddress);
+#endif
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE
+    void SetChildMulticastAddressMlrState(MlrState aFromState, MlrState aToState);
+    bool IsAddressMlrRegisteredByAnyChild(const Ip6::Address &aAddress)
+    {
+        return IsAddressMlrRegisteredByAnyChildExcept(aAddress, nullptr);
+    }
+    bool IsAddressMlrRegisteredByAnyChildExcept(const Ip6::Address &aAddress, const Child *aExceptChild);
```

Comment:
nit: declare as `const`?
```suggestion
    bool IsAddressMlrRegisteredByAnyChildExcept(const Ip6::Address &aAddress, const Child *aExceptChild) const;
```